module/zfs/dsl_pool.c

```diff
@@ -416,6 +429,12 @@ dsl_pool_close(dsl_pool_t *dp)
 	rrw_destroy(&dp->dp_config_rwlock);
 	mutex_destroy(&dp->dp_lock);
 	cv_destroy(&dp->dp_spaceavail_cv);
+
+	aggsum_fini(&dp->dp_wrlog_total);
+	for (int i = 0; i < TXG_SIZE; i++) {
+		aggsum_fini(&dp->dp_wrlog_pertxg[i]);
+	}
```

Comment:
How about adding an `ASSERT0` here for the `dp_wrlog_total` and per-txg `dp_wrlog_pertxg[i]` values to ensure the accounting is working as intended.

Reply:
Added.